lua/entities/gmod_wire_ledtape.lua

```diff
@@ -0,0 +1,341 @@
+		end
+		LocalToWorld_NoGarbage_Ents = {}
+	end
+
+	hook.Add("PostDrawOpaqueRenderables","LEDTapeCleanup",LocalToWorld_NoGarbage_End)
+
+	local function DrawBeams(width, scrollmul, mater, path, getColor, extravertex)
+
+		if not IsValid(path[1][1]) then return end
+
+		local scroll = 0
+		local beam   = render.AddBeam
+
+		local beam2     = extravertex and beam or function() end -- branchless programming ftw
+		local vertexnum = extravertex and 3 or 2
+
+		scrollmul = scrollmul / width -- scale this
+
+		render.SetMaterial(mater)
+		render.StartBeam(#path * vertexnum)
+
+			local node1 = path[1]
+
+			local pt1 = LocalToWorld_NoGarbage(node1[1], node1[2])
+
+			beam(pt1, width, scroll, getColor(node1))
+
+			for i = 2, #path do
+				local node2 = path[i]
+				local nodeEnt = node2[1]
+				if not IsValid(nodeEnt) then continue end
+				local nodeOffset = node2[2]
+
+				local pt2 = LocalToWorld_NoGarbage(nodeEnt, nodeOffset)
+				local distance = pt2:Distance(pt1) * scrollmul * 0.5
+
+				beam( pt1, width, scroll, getColor(node1))
+				scroll = scroll + distance
+				beam( pt2, width, scroll, getColor(node2))
+				beam2( pt2, width, scroll, getColor(node2)) -- add another point if extravertex is set, prevents some sprites from looking yucky
+
+				pt1 = pt2
+				node1 = node2
+			end
+
+			beam(pt1, width, scroll, getColor(node1))
+
+		render.EndBeam()
+		return pt1
+	end
+
+	function Wire_LEDTape.DrawShaded(width, scrollmul, mater, path)
+		return DrawBeams(width, scrollmul, mater, path, getLitNodeColor)
+	end
+
+	function Wire_LEDTape.DrawFullbright(width, scrollmul, color, mater, path, extravert)
+		return DrawBeams(width, scrollmul, mater, path, function(node) return color end, extravert)
+	end
+
+	function ENT:Initialize()
+
+		self:SharedInit()
+		self.ScrollMul = DEFAULT_SCALE
+
+		net.Start("LEDTapeData")
+			net.WriteEntity(self)
+			net.WriteBool(true) -- request full update
+		net.SendToServer()
+
+		if CLIENT then
+			local DrawShaded = Wire_LEDTape.DrawShaded
+			local DrawFullbright = Wire_LEDTape.DrawFullbright
+			hook.Add("PostDrawOpaqueRenderables", self, function()
+
+				if #self.Path < 2 then return end
+
+				if self.SpriteMaterial then
+
+					if self.Backlit then
+						DrawFullbright(self.Width, self.ScrollMul / 3, self.Color, self.BaseMaterial, self.Path, false)
+					else
+						DrawShaded(self.Width, self.ScrollMul / 3, self.BaseMaterial, self.Path)
+					end
+
+					DrawFullbright(self.Width * 3, self.ScrollMul, self.Color, self.SpriteMaterial, self.Path, not self.Connect)
+
+				else
+					DrawFullbright(self.Width, self.ScrollMul, self.Color, self.BaseMaterial, self.Path)
+				end
+
+			end)
+		end
+
+		self:SetOverlayText("LED Tape Controller")
+
+	end
+
+	function ENT:Think()
+		self.Color.r = self:GetNW2Int("LedTape_R")
+		self.Color.g = self:GetNW2Int("LedTape_G")
+		self.Color.b = self:GetNW2Int("LedTape_B")
+	end
+
+	net.Receive("LEDTapeData", function()
+
+		local controller = net.ReadEntity()
```

Comment:
You are not verifying that player owns said controller. Or that said controller is even an led tape.